File: 08_remove_vowels.py
"""
Task 8: Remove all vowels from a string.
Reads a line and prints it without vowels.
"""
import sys

def remove_vowels(s: str) -> str:
    vowels = set('aeiouAEIOU')
    return ''.join(ch for ch in s if ch not in vowels)

if __name__ == '__main__':
    s = sys.stdin.readline().rstrip('\n')
    print(remove_vowels(s))

# Time Complexity: O(n)
# Space Complexity: O(n)

File: 08_remove_vowels.cpp
#include <bits/stdc++.h>
using namespace std;

/*
Task 8: Remove all vowels from a string.
Reads a line and prints it without vowels.
*/

int main()
{
    string s;
    if (!getline(cin, s))
        return 0;
    string vowels = "aeiouAEIOU";
    for (char c : s)
        if (vowels.find(c) == string::npos)
            cout << c;
    cout << '\n';
    return 0;
}

// Time Complexity: O(n)
// Space Complexity: O(1)
